www/callbacks/coinbase.spt

```diff
@@ -0,0 +1,51 @@
+"""An endpoint to receive Coinbase's callbacks.
+
+Docs: https://developers.coinbase.com/docs/merchants/callbacks
+
+"""
+
+import json
+from decimal import Decimal as D
+
+from aspen import Response
+from gratipay.billing.exchanges import record_exchange, record_exchange_result
+from gratipay.models.exchange_route import ExchangeRoute
+from gratipay.models.participant import Participant
+
+[---]
+
+request.allow('POST')
+
+if request.qs.get('secret_key') != website.coinbase_secret_key:
+    raise Response(401)
+
+order = request.body.get('order', None)
+
+if order:
+    # Fetch the participant from the order's 'custom' attribute.
+    participant = Participant.from_id(int(order['custom']))
+
+    # Check if a route corresponding to the order ID already exists.
+    route = ExchangeRoute.from_address(participant, 'coinbase-payin', order['id'])
+
+    # If not, create one.
+    if not isinstance(route, ExchangeRoute):
+        route = ExchangeRoute.insert(participant, 'coinbase-payin', order['id'])
+
+    status = order['status']
+
+    # We're ignoring mispayments/expired orders here.
```

Comment:
Expired orders don't matter to us at all, but we could do more with mispayments.

We do show client side error messages like shown in the last screenshot in  https://github.com/gratipay/gratipay.com/pull/3317#issuecomment-91491278. Does sending an email sound good to you? The other options are listed in https://github.com/gratipay/gratipay.com/pull/3317#issuecomment-89835622, I think they're unnecessary and too much of work for now.